daemon/inspect_unix.go

```diff
@@ -21,7 +22,7 @@ func setPlatformSpecificContainerFields(container *container.Container, contJSON
 }
 
 // containerInspectPre120 gets containers for pre 1.20 APIs.
-func (daemon *Daemon) containerInspectPre120(name string) (*v1p19.ContainerJSON, error) {
+func (daemon *Daemon) containerInspectPre120(name string, version version.Version) (*v1p19.ContainerJSON, error) {
```

Comment:
How about NOT passing a `version` to `containerInspectPre120` and `containerInspect120`, but give them a fake version number? 
```
containerInspectPre120(name string)   // Make a assumption that the version is '1.19'
containerInspect120(name string)        // version is '1.20'
containerInspectCurrent(name string, size bool, version version.Version)   // pass the real version
```